docs/dev/data_validation.rst

```diff
@@ -0,0 +1,688 @@
+helpful output indicating what failed and why, in the same way as ``dbt_helper
+validate``
+
+Data validation in branch builds
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Depending on your computer, running the full ETL locally can be extremely time consuming
+and may run into memory limits. It's also easy to accidentally end up with local outputs
+that are the result of code from multiple different branches, and so may not be
+consistent with each other. If you're only altering a few tables, rematerializing them
+in Dagster and then running the specific dbt tests that apply to them and any tables
+downstream of them should work fine.
+
+Kicking off a branch build
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+But especially when we're doing big quarterly or annual updates, and dozens or hundreds
+of tables are changing simultaneously, it is helpful to be able to run the full ETL from
+scratch, run all of the data validation tests against the outputs, and use the results
+to update the test parameters (especially expected row counts) appropriately. This can
+be done by manually kicking off a PUDL deployment on your branch. To initiate a branch
+build, in the PUDL repo on GitHub go to `Actions
+<https://github.com/catalyst-cooperative/pudl/actions>`_ and select `build-deploy-pudl
+<https://github.com/catalyst-cooperative/pudl/actions/workflows/build-deploy-pudl.yml>`_.
+On the right hand side select Run Workflow and then select your branch in the dropdown
+and click the Run Workflow button. Shortly thereafter you should see a notification in
+the ``pudl-deployments`` channel in our Slack saying that the build has kicked off. It
+should take about 3 hours to complete. You can track its progress and watch the logs in
+the `Google Cloud Console
+<https://console.cloud.google.com/monitoring/dashboards/builder/992bbe3f-17e6-49c4-a9e8-8f1925d4ec24>`_.
+
+Getting fresh row counts from a branch build
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To catch unexpected changes to the data, we keep track of the expected number of rows in
+each data table we distribute. These expectations are stored in
+``dbt/seeds/etl_full_row_counts.csv`` and they can be updated using the ``dbt_helper``
+script based on the observed row counts in your local PUDL Parquet outputs. If you can't
+run the full ETL locally, the nightly builds / branch build also generate updated row
+count expectations. After a branch build completes, you can download the updated
+``etl_full_row_counts.csv`` file from the build outputs that are uploaded to
+``gs://builds.catalyst.coop/<build-id>/etl_full_row_counts.csv`` See the
+:doc:`nightly_data_builds` documentation for more details on accessing the nightly build
+outputs. Replace the ``etl_full_row_counts.csv`` in your local PUDL git repo with the
+one you've downloaded and use ``git diff`` to see what has changed. Make sure to review
+the row count changes closely to see if there's anything unexpected.
```

Comment:
Is this enough / the right information to make the branch/nightly build row counts useful to folks?

Reply:
Branch build rowcounts are definitely useful and this probably gives folks enough information to track them down.

Nightly build rowcounts I'm less sure about. Under what circumstances would the nightly build rowcounts differ from what's in the repo?